66_HLSLBxDFTests/tests.h

```diff
@@ -0,0 +1,704 @@
+#ifndef BXDFTESTS_TESTS_H
+#define BXDFTESTS_TESTS_H
+// cpp only tests
+
+#define GLM_ENABLE_EXPERIMENTAL
+#include <glm/gtx/hash.hpp>
+#include <unordered_map>
+#include <vector>
+#include <cmath>
+#include <format>
+#include <functional>
+
+#include "app_resources/tests_common.hlsl"
+
+// because unordered_map -- next time, do fixed size array of atomic offsets and linked lists (for readback and verification on cpu)
+template<class BxDF, bool aniso = false>
+struct TestBucket : TestBxDF<BxDF>
+{
+    using base_t = TestBxDFBase<BxDF>;
+    using this_t = TestBucket<BxDF, aniso>;
+    using traits_t = bxdf::traits<BxDF>;
+
+    void clearBuckets()
+    {
+        for (float y = -1.0f; y < 1.0f; y += stride)
+        {
+            for (float x = -1.0f; x < 1.0f; x += stride)
+            {
+                buckets[float32_t2(x, y)] = 0;
+            }
+        }
+    }
+
+    float bin(float a)
+    {
+        float diff = std::fmod(a, stride);
```

Comment:
nitpick, use the `hlsl` namespace's `tgmath` (or intrinsics, I'm sure that @Przemog1 added such a function) fmod function for this